pages/workoutlist.js

```diff
@@ -0,0 +1,164 @@
+
+function findExerciseById(exerciseId) {
+  return exercises.find((exercise) => exercise.id === exerciseId);
+}
+
+export default function WorkoutsList() {
+  const [showDetails, setShowDetails] = useState({});
+
+  const toggleDetails = (workoutId) => {
+    setShowDetails((prev) => ({
+      ...prev,
+      [workoutId]: !prev[workoutId],
+    }));
+  };
+
+  return (
+    <>
+      <StyledHeadline>Choose Your Workout</StyledHeadline>
+
+      <WorkoutList>
+        {workouts.map((workout) => {
+          const muscleGroupsSet = new Set();
+
+          workout.exercises.forEach((exercise) => {
+            const foundExercise = findExerciseById(exercise.exerciseId);
+
+            if (foundExercise) {
+              foundExercise.muscleGroups.forEach((muscleGroup) => {
+                muscleGroupsSet.add(muscleGroup);
+              });
+            }
+          });
+
+          const reducedMuscleGroups = Array.from(muscleGroupsSet);
+          const isDetailsVisible = showDetails[workout.id];
+          return (
+            <WorkoutCard key={workout.id}>
+              <h2>{workout.name}</h2>
+              <SpotlightHeading>Muscles In The Spotlight:</SpotlightHeading>
+              <MuscleGroupList>
+                {reducedMuscleGroups.map((muscle) => (
+                  <MuscleBadge key={muscle}>{muscle}</MuscleBadge>
+                ))}
+              </MuscleGroupList>
+              <ToggleButton onClick={() => toggleDetails(workout.id)}>
+                {isDetailsVisible ? "Show Less" : "Show More"}
+              </ToggleButton>
+              {isDetailsVisible && (
+                <ExerciseList>
+                  {workout.exercises.map((exercise) => {
+                    const foundExercise = findExerciseById(exercise.exerciseId);
+
+                    return (
+                      <ExerciseItem key={exercise.exerciseId}>
+                        <ExerciseName>{foundExercise.name}</ExerciseName>
+                        <ExerciseDetails>Sets: {exercise.sets}</ExerciseDetails>
+                        <ExerciseDetails>Reps: {exercise.reps}</ExerciseDetails>
+                      </ExerciseItem>
+                    );
+                  })}
+                </ExerciseList>
+              )}
+            </WorkoutCard>
+          );
+        })}
+      </WorkoutList>
+    </>
+  );
+}
+
+const WorkoutList = styled.div`
+  width: 100%;
+  max-width: 1200px;
+  display: flex;
+  justify-content: center;
+  padding: 1rem;
+  flex-direction: column;
+`;
+
+const WorkoutCard = styled.div`
+  position: relative;
+  border-radius: 10px;
+  overflow: hidden;
+  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
+  margin-bottom: 2rem;
+  padding: 1rem;
+  background: #ffffff;
+  text-align: center;
+`;
+
+const StyledHeadline = styled.h2`
+  background: #fff;
+  padding: 0.5rem;
+  margin: 0;
+  text-align: center;
+`;
+
+const ToggleButton = styled.button`
+  background: #e67e22;
+  color: #ffffff;
```

Comment:
No must but nice for consistency

```suggestion
  color: #fff;
```